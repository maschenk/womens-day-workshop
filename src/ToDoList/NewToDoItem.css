.add-button {
  display: inline-flex;
  box-sizing: border-box;
  height: 2rem;
  margin-left: 20px;
  min-width: 7.5rem;
  justify-content: center;
  align-items: center;
  padding: 0 1rem;
  border: none;
  border-radius: .25rem;
  background-color: #ff790f;
  color: #fff;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  letter-spacing: .025rem;
  transition: box-shadow .1s,background-color .1s;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
}