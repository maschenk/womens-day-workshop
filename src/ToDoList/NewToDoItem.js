import React from 'react';

class NewToDoItem extends React.Component {
  constructor() {
    super();
    this.state = { text: '' };

    this.handleAdd = this.handleAdd.bind(this);
    this.updateText = this.updateText.bind(this);
  }

  handleAdd(event) {
    event.preventDefault();
    const { text: newItemText } = this.state;
    const { onAdd } = this.props;

    onAdd(newItemText);

    this.setState({ text: '' });
  }

  updateText(event) {
    event.preventDefault();
    this.setState({ text: event.target.value });
  }

  render() {
    return (
      <div data-testid="new-to-do-item">
        <input
          type="text"
          onInput={this.updateText}
          value={this.state.text}
          data-testid="item-input"
        />

        <button type="button" onClick={this.handleAdd} data-testid="add-button">
          Add To Do
        </button>
      </div>
    );
  }
}

export default NewToDoItem;
