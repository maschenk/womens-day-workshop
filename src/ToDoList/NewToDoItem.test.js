import { render, fireEvent } from '@testing-library/react';
import NewToDoItem from './NewToDoItem';

const onAddMock = jest.fn();

test('should render', () => {
  const { container } = render(<NewToDoItem onAdd={onAddMock} />);
  expect(container).toMatchSnapshot();
});

test('should add item', () => {
  const component = render(<NewToDoItem onAdd={onAddMock} />);

  const inputField = component.getByTestId('item-input');

  fireEvent.input(inputField, { target: { value: 'add a test item' } });
  expect(inputField.value).toBe('add a test item');

  const addButton = component.getByTestId('add-button');

  fireEvent.click(addButton);
  expect(onAddMock).toHaveBeenCalled();
});
