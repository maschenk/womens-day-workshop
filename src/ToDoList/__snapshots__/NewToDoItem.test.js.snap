// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    data-testid="new-to-do-item"
  >
    <input
      data-testid="item-input"
      type="text"
      value=""
    />
    <button
      data-testid="add-button"
      type="button"
    >
      Add a new Item
    </button>
  </div>
</div>
`;
